code-conventions.md

```diff
@@ -1,36 +1,44 @@
-# Coding Conventions
+# Code Conventions
 
 ## Security
-(this is very important)
+Keep in mind that your code will be running on hundreds or thousands of
+servers; security matters. Following these guidelines is *vital* for all new
+code. Contributors should also see the dedicated
+[security document](security.md).
+
 * **NO MALICIOUS CODE** - I think you already knew this
   ```java
   Runtime.getRuntime().exec("rm -rf /"); // Really BAD
   ```
-* **DO NOT** submit private information (passwords, etc.) to the Git
+* **DO NOT** submit secrets (passwords, etc.) to the Git
   ```java
   String myApiToken = "f6dd0c930676da532136fbef805ef1553ebaca83"; // Really BAD
   // Though this is not a real token - can you find what it is?
   ```
+  - If this happens and somehow gets past code review, you'll have to change
+    your secrets; we are generally unwilling to 'force push' anything
 * Client or server crashes/freezes shouldn't be easy to create from scripts
   ```java
   if (matchedPattern == 0)
       wait(1000); // BAD
   ```
-* Obvious bad ideas (such as MD5 hashing) should not look safe to scripters
+  - This is not to say that you can't currently crash the client, but avoid
+    adding any more ways to cause trouble
+  - Be especially wary of things people might accidentally use
+* Obviously bad ideas (such as MD5 hashing) should not look safe to scripters
   ```java
   MessageDigest md5 = MessageDigest.getInstance("MD5"); // BAD
   // ... although this is a snippet from Skript source code. Legacy stuff!
   ```
+  - Make it clear when scripts are doing something insecure or dangerous
+  - Exception: consider twice before breaking API exposed to scripts
 * File or network IO, for now, should not be accessible from scripts
   ```java
   String fileName = file.getSingle(e); // Bad
   ```
 * Code contributed must be licensed under GPLv3, by *you*
   - Third party code under compatible licenses may be accepted in some cases
-
-And, additionally:
-* Everyone with direct push access must use two factor authentication
-  - https://xkcd.com/936/
+  - We'll forward any complaints about code licensing to people who added it
 
 ## Formatting
```

Comment:
Suggested by @DeltaRays, we should have another statement here to inform people that they should not squeeze multiple lines of code on a single line.